src/bindings/js/node/include/core_set_property_args.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <napi.h>
+
+#include "openvino/runtime/core.hpp"
+
+/**
+ * @brief This struct retrieves data from Napi::CallbackInfo.
+ */
+struct CoreSetPropertyArgs {
+    std::string device_name;
+    ov::AnyMap parameters;
+
+    CoreSetPropertyArgs() {}
+    CoreSetPropertyArgs(const Napi::CallbackInfo& info) {
+        const size_t args_length = info.Length();
+
+        if (!is_valid_input(args_length, info))
+            throw std::runtime_error("Invalid arguments of set_property function");
+
+        if (args_length > 1) device_name = info[0].ToString();
+        
+        Napi::Object parameters = info[args_length > 1 ? 1 : 0].ToObject();
+        const auto& keys = parameters.GetPropertyNames();
+
+        for (uint32_t i = 0; i < keys.Length(); ++i) {
+            auto property_name = static_cast<Napi::Value>(keys[i]).ToString().Utf8Value();
+            
+
+            ov::Any any_value = js_to_any(info, parameters.Get(property_name));
+
+            this->parameters.insert(std::make_pair(property_name, any_value));
+        }
+    }
+
+    bool is_valid_input(size_t args_length, const Napi::CallbackInfo& info) {
```

Comment:
What is this function validating? Length of input? If `params/device_name` were passed?

Reply:
Arguments passed from js part)